sqldelight-runtime/src/main/java/com/squareup/sqldelight/db/SqlResultSet.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.db
+
+interface SqlResultSet {
+  fun next(): Boolean
+  fun getString(index: Int): String
+  fun getLong(index: Int): Long
+  fun getBytes(index: Int): ByteArray
+  fun getFloat(index: Int): Float
+  fun close()
+}
+
+inline fun <R> SqlResultSet.use(block: (SqlResultSet) -> R): R {
```

Comment:
this can’t just be closeable?

Reply:
errr. AutoCloseable

Reply:
its a java dependency :(

Reply:
so `.use()` isn’t available in Kotlin? Weak sauce

Reply:
More that there's no `Closeable` abstraction.

Reply:
More that there's no `Closeable` abstraction.